src/QuadForm/indefiniteLLL.jl

```diff
@@ -0,0 +1,357 @@
+[4 5]
+[7 8]
+=#
+function _vecextract(M::MatElem,x::Int64,y::Int64)
+  x_bin = digits(x,base = 2)
+  y_bin = digits(y, base = 2)
+
+  list_x = [i for i = 1:length(x_bin) if x_bin[i] == 1]
+  list_y = [i for i = 1:length(y_bin) if y_bin[i] == 1]
+
+  return M[list_x , list_y]
+
+end
+
+function _vecextract(M::MatElem, x::Union{Int64,Vector{Int64}},y::Int64)
+  y_bin = digits(y, base = 2)
+  list_y = [i for i = 1:length(y_bin) if y_bin[i] == 1]
+
+  return M[x , list_y]
+
+end
+
+################################################################################
+#                           linear algebra
+################################################################################
+
+#=
+    _mathnf(A::MatElem) -> MatElem, MatElem
+
+Given a rectangular matrix $A$ of dimension $nxm$. Computes the Hessian matrix $H$ of dimension $nxm$ 
+and the unimodular transformation matrix $U$ such that $AU$ = $H$.
+=#
+function _mathnf(A::MatElem)
+  H, U = hnf_with_transform(reverse_cols(transpose(A)))
+  H = reverse_rows(reverse_cols(transpose(H)))
+  U = reverse_cols(transpose(U))
+
+  return H, U
+end
+
+#=
+    _complete_to_basis(v::MatElem; redflag::Bool = false) -> MatElem
+
+Given a rectangular matrix $nxm$ with $n$ != $m$ and redflag = 0.
+Computes a unimodular matrix with the last column equal to the last column of $v$.
+If redflag = 1, it LLL-reduce the $n$-$m$ first columns if $n$ > $m$.
+=#
+function _complete_to_basis(v::MatElem, redflag::Bool = false)
+  
+  n = nrows(v) #Number of rows
+  m = ncols(v) #Number of cols
+
+  if(n == m) #return nxn matrices
+    return v
+  end
+
+  U = inv(transpose(_mathnf(transpose(v))[2]))
+
+  if(n == 1 || redflag == false)
+    return U
+  end
+
+  re = U[:,1:n-m]
+  re = transpose(lll_with_transform(transpose(re))[2])
+
+  l = diagonal_matrix(re,one(parent(v[1:m,1:m])))
+  re = U*l
+
+  return re
+end
+
+#=
+    _ker_mod_p(M::MatElem,p) -> Int, MatElem
+
+Computes the kernel of the given matrix $M$ mod $p$.
+It returns [$rank$,$U$], where $rank$ = dim (ker M mod p) and $U$ in $GL_n$(Z),
+The first $rank$ columns of $U$ span the kernel.; check::Bool = false
+=#
+function _ker_mod_p(M::MatElem,p)
+  R = parent(M[1,1])
+  rk, k = kernel(change_base_ring(ResidueRing(R,p),M))
+  U = _complete_to_basis(lift(k[:,1:rk]))
+  reverse_cols!(U)
+
+  return rk, U
+end
+
+################################################################################
+#                           Quadratic Equations
+################################################################################
+
+#=
+    _quad_form_solve_triv(G::MatElem{fmpz}; base::Bool = false) -> MatElem{fmpz}
+
+Trying to solve $G$ = 0 with small coefficients. Works if $det$($G$) = 1, dim <= 6
+and $G$ is LLL-reduced. Return [$G$,$I_n$] if no solution is found.
+Exit with a norm 0 vector if one such is found.
+If base = 1 and a norm 0 vector is obtained, returns $transpose$($H$)*$G$*$H$, 
+$H$, $sol$ where $sol$ is of norm 0 vand is the first column of $H$.
+=#
```

Comment:
"Trying to solve $G$ = 0 ... " does not have much sense: $G = 0$ or $G \neq 0$ is quite obvious. You could say something like "Try to compute a non-zero vector in the kernel of `G` with small coefficients", for instance. 
"Works if ... " should be more like " `G` must be a square matrix of determinant 1 and dimension at most 6" and therefore, put input checks in your function. You could avoid the "$G$ is LLL-reduced" since we don't have a proper way to test it, and your function does not check it. However, you could say something like " `G` should be the output of the function(bla)" if you want to stress on the fact that `G` has been reduced somehow somewhere. 
Again, here your function should return something more even though you don't have any isotropic vector found. You can return the empty vector, like  `fmpz[]`. Moreover, you don't have to use square brackets in the doc: "Return [$G$, $I_n$]" could be turned into "Return $M, H, sol$ where .... is ... if ..., ... otherwise.". Consistency on the output is important for type stability and to having to avoid checking which kind of output you have obtained.
The same here, if `base == false`, you should not only return `sol`, but also two matrices, in this context it would mean `G` and the identity matrix of the correct size.